operator/pkg/ciliumendpointbatch/manager.go

```diff
@@ -0,0 +1,422 @@
+	cepName := getCepNameFromCCEP(cep)
+	log.WithFields(logrus.Fields{
+		"cep-name": cepName,
+	}).Debug("Remove CEP from local cache")
+	if cebName, ok := c.cacheCepMap.get(cepName); ok {
+		ceb, err := c.cacheCeb[cebName]
+		if err {
+			if ceb == nil || ceb.ceb == nil {
+				log.Info("Valid cep Mapping, but no matching CEP in CEB")
+				return nil
+			}
+		}
+
+		for i, ep := range ceb.ceb.Endpoints {
+			if getCepNameFromCCEP(&ep) == getCepNameFromCCEP(cep) {
+				ceb.backendMutex.Lock()
+				// Insert deleted CoreCEP in removedCeps
+				ceb.removedCeps[getCepNameFromCCEP(&ep)] = struct{}{}
+				ceb.ceb.Endpoints =
+					append(ceb.ceb.Endpoints[:i],
+						ceb.ceb.Endpoints[i+1:]...)
+				ceb.backendMutex.Unlock()
+				break
+			}
+		}
+		log.WithFields(logrus.Fields{
+			"ceb-name":  cebName,
+			"cep-name":  cepName,
+			"cep-count": len(ceb.ceb.Endpoints),
+		}).Debug("Removed cep from ceb")
+
+		c.cacheCepMap.deleteCep(cepName)
+
+		if len(ceb.ceb.Endpoints) == 0 {
+			log.WithFields(logrus.Fields{
+				"ceb-name": cebName,
+			}).Debug("Remove CEB from local cache")
+			c.cebsToSync.updateAggregator(cebName, CebDelete)
+			return nil
+		}
+
+		c.cebsToSync.updateAggregator(cebName, CebLazyUpdate)
+		return nil
+	}
+
+	return nil
+}
+
+// Return the total number of CEPs in cluster
+func (c *cebManagerFcfs) getTotalCepCount() int {
+	cnt := 0
+	for _, ceb := range c.cacheCeb {
+		cnt += len(ceb.ceb.Endpoints)
+	}
+	return cnt
+}
+
+// Return the total number of CEPs in ceb
+func (c *cebManagerFcfs) getCepCountInCeb(cebName string) (cnt int) {
+	if ceb, _ := c.cacheCeb[cebName]; ceb != nil {
+		cnt = len(ceb.ceb.Endpoints)
+	}
+	return
+}
+
+// Return the total count of CEB
+func (c *cebManagerFcfs) getCebCount() int {
+	return len(c.cacheCeb)
+}
+
+// Return list of Core CEPs
+func (c *cebManagerFcfs) getAllCeps() map[string]*cilium_v2.CoreCiliumEndpoint {
+	ceps := make(map[string]*cilium_v2.CoreCiliumEndpoint)
+	for _, ceb := range c.cacheCeb {
+		ceb.backendMutex.RLock()
+		for _, cep := range ceb.ceb.Endpoints {
+			ceps[getCepNameFromCCEP(&cep)] = cep.DeepCopy()
+		}
+		ceb.backendMutex.RUnlock()
+	}
+
+	return ceps
+}
+
+// Return the list of removed Core CEPs
+func (c *cebManagerFcfs) getRemovedCeps(cebName string) map[string]struct{} {
+	cepNames := make(map[string]struct{})
+	if ceb, _ := c.cacheCeb[cebName]; ceb != nil {
+		ceb.backendMutex.RLock()
+		for cepName := range ceb.removedCeps {
+			cepNames[cepName] = struct{}{}
+		}
+		ceb.backendMutex.RUnlock()
+	}
+
+	return cepNames
+}
+
+// After successful sync with api-server, delete removed ceps in a CEB.
+// If no more CEPs are packed in CEB, Delete the CEB in next Delete SYNC.
+func (c *cebManagerFcfs) clearRemovedCeps(cebName string, remCeps map[string]struct{}) {
+
+	var ceb *ceBatch
+	// Check if CEB exists in local cache
+	if ceb, _ = c.cacheCeb[cebName]; ceb == nil {
+		return
+	}
+
+	// Delete removed CEPs from caches.
+	for cn := range remCeps {
```

Comment:
shouldn't this be
```
for cn, _ := range remCeps {
```

Reply:
Map can be traversed by keys, as long as we don't care about values.